compiler/ballerina-parser/src/main/java/io/ballerinalang/compiler/internal/parser/BallerinaParser.java

```diff
@@ -4976,13 +4989,25 @@ private STNode parseRemoteKeyword() {
     private STNode parseObjectField(STNode metadata, STNode visibilityQualifier, boolean isObjectTypeDesc) {
         STToken nextToken = peek();
         if (nextToken.kind != SyntaxKind.READONLY_KEYWORD) {
-            STNode readonlyQualifier = STNodeFactory.createEmptyNode();
+            STNode finalQualifier = STNodeFactory.createEmptyNode();
```

Comment:
I think we can simplify this entire method. Since `readonly` is no longer a qualifier for the field, we don't need to resolve ambiguities. Can simply do like:
```ballerina
parseQualifiers();   // includes both visibility-qualifiers and final-qualifier
parseTypeDescriptor();
parseFieldName();
...
```

Reply:
All logic underneath this if block is not required. Kept it to give better error msg if user input readonly as usual. 
Will remove and simplify.

Reply:
Can we pass both visibility qualifier and final in the same method? Thing is visibility qualifier is used by method-defn/method-decl as well. And Final is only needed for object-fields. So kept the parsing in two stages. Plz review new changes.